backend/zkppSalt.js

```diff
@@ -0,0 +1,252 @@
+import { blake32Hash } from '~/shared/functions.js'
+import { timingSafeEqual } from 'crypto'
+import nacl from 'tweetnacl'
+import sbp from '@sbp/sbp'
+import { boxKeyPair, encryptContractSalt, hashStringArray, hashRawStringArray, hash, parseRegisterSalt, randomNonce, computeCAndHc, base64urlToBase64, base64ToBase64url } from '~/shared/zkpp.js'
+
+// TODO HARDCODED VALUES
+// These values will eventually come from server the configuration
+const recordPepper = 'pepper'
+const recordMasterKey = 'masterKey'
+const challengeSecret = 'secret'
+const registrationSecret = 'secret'
```

Comment:
We don't want to have server admins dealing with anything related to cryptography unless for some reason they must and there's literally no way around it.

So what are these values and how can we avoid making the admin set them via configs?

Reply:
> So what are these values and how can we avoid making the admin set them via configs?

These are secrets the server uses internally for deriving certain values that must be opaque to the client. As these values need to be persistent, I can't come up with a way of avoiding them entirely (although a solution could be for them to be set to some default value).9

> `const recordPepper = 'pepper'`

Perhaps the least security-critical value. Used to derive a record ID which clients / 3rd parties cannot guess for storing the ZKPP

> `const recordMasterKey = 'masterKey'`

Used to encrypt ZKPP records. If this value is leaked, 3rd parties with access to the database could defeat the protocol.

> `const challengeSecret = 'secret'`

If this value is known, the protocol can be defeated by providing valid answers to challenges.

> `const registrationSecret = 'secret'`

Similarly, but for registration.

Reply:
So what about making them random and storing them in the key-val db?

Reply:
Sure, it could be a part of the installation process, and could be automated.

Reply:
I'm going to use this convention to mark threads as needing their commentary transferred (as comments) to the code.

➡️ code comments

Reply:
Note from call: we can get rid of `recordPepper`, just use `private/rid/${contractID}`

`challengeSecret` is the key to the keyed hash function in step 3 of the protocol.